### PR TITLE
MINIFICPP-2418 Drop JNI extension

### PR DESCRIPTION
**Pull request by martinzink**
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: docker/Dockerfile
Comment (lordgamez) [inline docker/Dockerfile:1]: I think the default JDK installation and setting java home in the path can also be removed from all of the dockerfiles. Probably it should only be kept the same way it is done in `docker/centos/Dockerfile`
```
if echo "$MINIFI_OPTIONS" | grep -q "ENABLE_SFTP=ON" && [ "${DOCKER_SKIP_TESTS}" == "OFF" ]; then yum -y install java-1.8.0-openjdk maven; fi
```

 Also I think java path setting can also be removed from DockerVerify.sh

File: extensions/jni/nifi-framework-jni/src/main/java/org/apache/nifi/nar/JniUnpacker.java
[the remark this answers is not in the document]
Comment (martinzink) [reply extensions/jni/nifi-framework-jni/src/main/java/org/apache/nifi/nar/JniUnpacker.java:1]: good catch, removed them in https://github.com/apache/nifi-minifi-cpp/pull/1829/commits/75ff0188305cdd0edd6bc453e1d559ded513e42d#diff-dfb14fbb9e7d095209ec4cfd621069437bf9c442ff9de9d4ce889781bd0fefcfL10-L18

File: Windows.md [hunk 1]
```diff
@@ -134,7 +127,7 @@ A basic working CMake configuration can be inferred from the `win_build_vs.bat`.
...
+cmake -G "Visual Studio 17 2022" -A x64 -DMINIFI_INCLUDE_VC_REDIST_MERGE_MODULES=OFF -DTEST_CUSTOM_WEL_PROVIDER=OFF -DENABLE_SQL=OFF -DMINIFI_USE_REAL_ODBC_TEST_DRIVER=OFF -DCMAKE_BUILD_TYPE_INIT=Release -DCMAKE_BUILD_TYPE=Release -DWIN32=WIN32 -DENABLE_LIBRDKAFKA=OFF -DENABLE_AWS=OFF -DENABLE_PDH= -DENABLE_AZURE=OFF -DENABLE_SFTP=OFF -DENABLE_SPLUNK= -DENABLE_GCP= -DENABLE_NANOFI=OFF -DENABLE_OPENCV=OFF -DENABLE_PROMETHEUS=OFF -DENABLE_ELASTICSEARCH= -DUSE_SHARED_LIBS=OFF -DENABLE_CONTROLLER=ON -DENABLE_BUSTACHE=OFF -DENABLE_COAP=OFF -DENABLE_ENCRYPT_CONFIG=OFF -DENABLE_GPS=OFF -DENABLE_LUA_SCRIPTING=OFF -DENABLE_MQTT=OFF -DENABLE_OPC=OFF -DENABLE_OPENWSMAN=OFF -DENABLE_OPS=OFF -DENABLE_PCAP=OFF -DENABLE_PYTHON_SCRIPTING= -DENABLE_SENSORS=OFF -DENABLE_USB_CAMERA=OFF -DBUILD_ROCKSDB=ON -DUSE_SYSTEM_UUID=OFF -DENABLE_LIBARCHIVE=ON -DENABLE_WEL=ON -DMINIFI_FAIL_ON_WARNINGS=OFF -DSKIP_TESTS=OFF ..
```
[the remark this answers is not in the document]
Comment (martinzink) [reply Windows.md]: good catch, it snuck back in during rebase
https://github.com/apache/nifi-minifi-cpp/pull/1829/commits/75ff0188305cdd0edd6bc453e1d559ded513e42d#diff-2e14d9d3ed6506710cd2e3b1ad21300f81bb66771bdb79395eba86b9eae33a42R130

File: docker/Dockerfile
Comment (martinzink) [reply docker/Dockerfile:1]: 👍 https://github.com/apache/nifi-minifi-cpp/pull/1829/commits/75ff0188305cdd0edd6bc453e1d559ded513e42d#diff-f34da55ca08f1a30591d8b0b3e885bcc678537b2a9a4aadea4f190806b374ddcL44-L46

File: extensions/jni/ExecuteJavaControllerService.cpp
[the remark this answers is not in the document]
Comment (martinzink) [reply extensions/jni/ExecuteJavaControllerService.cpp:1]: good catch, https://github.com/apache/nifi-minifi-cpp/pull/1829/commits/75ff0188305cdd0edd6bc453e1d559ded513e42d#diff-dccd4a349e00f087de9bd61ae264578572acc308f6d145e15d658866ffcd6204R57-R58

**Review by szaszm — APPROVED** [review]

**Review by lordgamez — APPROVED** [review]

**Pull request closed without merging**